src/presentation/http/router/note.ts

```diff
@@ -164,14 +164,13 @@ const NoteRouter: FastifyPluginCallback<NoteRouterOptions> = (fastify, opts, don
     /**
      * Get all tools used in the note
      */
-    const noteToolsNames = new Set<string>();
+    const noteToolsIds : EditorTool['id'][] = [];
 
-    note.content.blocks.forEach((block: { type: string }) => {
-      noteToolsNames.add(block.type);
+    Array.from(note.tools).forEach((tool) => {
+      /* for each tools there would be only one value (toolId) */
+      noteToolsIds.push(Object.values(tool)[0]);
     });
```

Comment:
You can use [reduce()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) here

```suggestion
    const noteToolsIds = note.tools.reduce((result, tool) => {
      result.push(Object.values(tool)[0]);
      
      return result;
    }, [])
```